src/main/ruby/core/truffle/kernel_operations.rb

```diff
@@ -14,5 +14,29 @@ def self.define_hooked_variable(name, getter, setter, defined = proc { 'global-v
       defined = Truffle::Graal.always_split(defined) if defined.arity == 1
       define_hooked_variable_with_is_defined(name, getter, setter, defined)
     end
+
+    def self.define_read_only_global(name, getter)
+      setter = -> _ { raise NameError, "#{name} is a read-only variable." }
+      define_hooked_variable(name, getter, setter)
+    end
   end
 end
+
+load_path = []
+loaded_features = []
+
+Truffle::KernelOperations.define_read_only_global(:$LOAD_PATH, -> { load_path })
+Truffle::KernelOperations.define_read_only_global(:$LOADED_FEATURES, -> { loaded_features })
```

Comment:
Reading those will not be constant and go through a couple indirections.
How about making them Ruby constants on `Truffle::KernelOperations`?